### PR TITLE
Possible Commands

### LINKED ISSUE #4: Autogenerate bash-completion script from all sub commands
It would be handy to autogenerate bash completion script from all subcomands that are defined for application. For start, only command names could be completed.

### PR DESCRIPTION
**Pull request by vkrizan**
Changelog:
* fixed to copy namespace before usage 
* move subcommand getting and parsing to separate function
* raise `CommandUnrecognizedArgs` for subcommands
* function to return possible (sub)commands for provided cli arguments

Closes #4

### CODE REVIEW

File: smclip/commands.py [hunk 1]
```diff
@@ -379,6 +382,21 @@ def invoke_default(self, raw_args):
     def get_subcmd_real_name(self, subcmd_cls):
         return self._subcmd_names.get(subcmd_cls)
 
+    def commands_to_be_used(self, raw_args):
```
Comment (isimluk) [inline smclip/commands.py]: a little documentation wouldn't hurt, would it?

**Comment by isimluk** [comment]
- Too much stuff happening in f98062828e0a0d54f473820b16feb0132965496e, that makes it hard for prudent review.

**Comment by isimluk** [comment]
This PR is a stream of thoughts. Commit no 8 amends changes in commit no 4, etc...

It would be easier for reviewer if you could polish it a bit using `git rebase -i`.

File: smclip/commands.py (hunk 1 above)
Comment (vkrizan) [reply smclip/commands.py]: Added in fe81533. thx

**Pull request merged**